ta/os_test/os_test.c

```diff
@@ -35,6 +35,7 @@
 #include "os_test.h"
 #include "testframework.h"
 #include "test_float_subj.h"
+#include "undef_instr.h"
```

Comment:
OK for generally preferring `.S` over inline assembly. Still, creating 3 source files for one instruction seems a bit excessive. Actually, the Linux kernel CodingStyle discourages the use of inline assembly for complex functions only:

> Chapter 19:  Inline assembly
>
> In architecture-specific code, you may need to use inline assembly to interface with CPU or platform functionality. Don't hesitate to do so when necessary. [...]
> Large, non-trivial assembly functions should go in .S files, with corresponding C prototypes defined in C header files.

If inline assembly is a concern, why not keep the current code? Simply adjust the value of the undefined instruction for A32 and A64, add a comment stating the intent or rename `my_undef` to `undef_instr` and voilà.